File: app/src/main/java/com/philsoft/metrotripper/utils/ui/Ui.java
package com.philsoft.metrotripper.utils.ui;

import android.app.Activity;
import android.app.Fragment;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.util.DisplayMetrics;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


public class Ui {

    @SuppressWarnings("unchecked")
    public static <T extends Fragment> T findFrag(Activity activity, String tag) {
        return (T) activity.getFragmentManager().findFragmentByTag(tag);
    }

    @SuppressWarnings("unchecked")
    public static <T extends View> T findView(Activity activity, int id) {
        return (T) activity.findViewById(id);
    }

    @SuppressWarnings("unchecked")
    public static <T extends View> T findView(View enclosingView, int id) {
        return (T) enclosingView.findViewById(id);
    }

    public static TextView setText(View enclosingView, int textViewId, String text) {
        TextView textView = findView(enclosingView, textViewId);
        textView.setText(text);
        return textView;
    }

    public static TextView setText(Activity activity, int textViewId, String text) {
        TextView textView = findView(activity, textViewId);
        textView.setText(text);
        return textView;
    }

    public static Bitmap createBitmapFromView(Context context, View view) {
        DisplayMetrics displayMetrics = new DisplayMetrics();
        ((Activity) context).getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
        view.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT));
        view.measure(displayMetrics.widthPixels, displayMetrics.heightPixels);
        view.layout(0, 0, displayMetrics.widthPixels, displayMetrics.heightPixels);
        view.buildDrawingCache();
        Bitmap bitmap = Bitmap.createBitmap(view.getMeasuredWidth(), view.getMeasuredHeight(), Bitmap.Config.ARGB_8888);

        Canvas canvas = new Canvas(bitmap);
        view.draw(canvas);

        return bitmap;
    }

    public static Bitmap createBitmapFromDrawableResource(Context context, int widthOffset, int heightOffset, int drawableResource) {
        Drawable d = context.getResources().getDrawable(drawableResource);
        BitmapDrawable bd = (BitmapDrawable) d.getCurrent();
        Bitmap b = bd.getBitmap();
        return Bitmap.createScaledBitmap(b, b.getWidth() + widthOffset, b.getHeight() + heightOffset, false);
    }

    public static Bitmap createBitmapFromLayoutResource(Activity activity, int layoutResource) {
        View view = activity.getLayoutInflater().inflate(layoutResource, null, false);
        return createBitmapFromView(activity, view);
    }

}

File: app/src/main/java/com/philsoft/metrotripper/utils/ui/Ui.kt
package com.philsoft.metrotripper.utils.ui

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import android.graphics.Point
import android.graphics.drawable.BitmapDrawable
import android.view.WindowManager
import com.philsoft.metrotripper.utils.createBitmap


object Ui {
    fun createBitmapFromDrawableResource(context: Context, widthOffset: Int, heightOffset: Int, drawableResource: Int): Bitmap {
        val d = context.resources.getDrawable(drawableResource)
        val bd = d.current as BitmapDrawable
        val b = bd.bitmap
        return Bitmap.createScaledBitmap(b, b.width + widthOffset, b.height + heightOffset, false)
    }

    fun createBitmapFromLayoutResource(activity: Activity, layoutResource: Int): Bitmap {
        val view = activity.layoutInflater.inflate(layoutResource, null, false)
        return view.createBitmap()
    }

    fun getCurrentScreenHeight(context: Context): Int {
        val size = getDisplaySize(context)
        return size.y
    }

    private fun getDisplaySize(context: Context): Point {
        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val display = wm.defaultDisplay
        val size = Point()
        display.getSize(size)
        return size
    }
}
